pkg/rules/prometheus_test.go

```diff
@@ -0,0 +1,178 @@
+					File:                              fmt.Sprintf("%s/../../examples/alerts/alerts.yaml", curr),
+					Rules:                             []*rulespb.Rule{someAlert, someAlert},
+					Interval:                          60,
+					DeprecatedPartialResponseStrategy: storepb.PartialResponseStrategy_WARN, PartialResponseStrategy: storepb.PartialResponseStrategy_WARN,
+				},
+				{
+					Name:                              "thanos-store.rules",
+					File:                              fmt.Sprintf("%s/../../examples/alerts/alerts.yaml", curr),
+					Rules:                             []*rulespb.Rule{someAlert, someAlert, someAlert, someAlert},
+					Interval:                          60,
+					DeprecatedPartialResponseStrategy: storepb.PartialResponseStrategy_WARN, PartialResponseStrategy: storepb.PartialResponseStrategy_WARN,
+				},
+				{
+					Name:                              "thanos-bucket-replicate.rules",
+					File:                              fmt.Sprintf("%s/../../examples/alerts/rules.yaml", curr),
+					Rules:                             []*rulespb.Rule{},
+					Interval:                          60,
+					DeprecatedPartialResponseStrategy: storepb.PartialResponseStrategy_WARN, PartialResponseStrategy: storepb.PartialResponseStrategy_WARN,
+				},
+				{
+					Name:                              "thanos-query.rules",
+					File:                              fmt.Sprintf("%s/../../examples/alerts/rules.yaml", curr),
+					Rules:                             []*rulespb.Rule{someRecording, someRecording, someRecording, someRecording, someRecording},
+					Interval:                          60,
+					DeprecatedPartialResponseStrategy: storepb.PartialResponseStrategy_WARN, PartialResponseStrategy: storepb.PartialResponseStrategy_WARN,
+				},
+				{
+					Name: "thanos-receive.rules", File: fmt.Sprintf("%s/../../examples/alerts/rules.yaml", curr),
+					Rules:                             []*rulespb.Rule{someRecording, someRecording, someRecording, someRecording, someRecording, someRecording},
+					Interval:                          60,
+					DeprecatedPartialResponseStrategy: storepb.PartialResponseStrategy_WARN, PartialResponseStrategy: storepb.PartialResponseStrategy_WARN,
+				},
+				{
+					Name:                              "thanos-store.rules",
+					File:                              fmt.Sprintf("%s/../../examples/alerts/rules.yaml", curr),
+					Rules:                             []*rulespb.Rule{someRecording, someRecording, someRecording, someRecording},
+					Interval:                          60,
+					DeprecatedPartialResponseStrategy: storepb.PartialResponseStrategy_WARN, PartialResponseStrategy: storepb.PartialResponseStrategy_WARN,
+				},
+			},
+			// TODO(bwplotka): Potentially add more cases.
+		},
+	} {
+		t.Run("", func(t *testing.T) {
+			srv := &rulesServer{ctx: ctx}
+			err = promRules.Rules(&rulespb.RulesRequest{}, srv)
+			if tcase.expectedErr != nil {
+				testutil.NotOk(t, err)
```

Comment:
Minor thing but this is unnecessary since we check for equality on the next line and `tcase.expectedErr` is guaranteed not to be `nil`. BTW, great test :smile: